enhancements/20250606-sidecar-support.md

```diff
@@ -0,0 +1,92 @@
+# Sidecar support
+
+Enable Kubvirt sidecar support to allow fine tuning of the Kubevirt generated domain definitions
+
+## Summary
+
+This enhancement aims to provide a generic way to tweak Kubevirt generated domain definitions to allow fine tuning of limited set of settings.
+
+### Related Issues
+
+https://github.com/harvester/harvester/issues/5619
+
+## Motivation
+
+### Goals
+
+To provide a generic way to leverage Harvester built and 3rd party sidecars with Harvester
+
+### Non-goals [optional]
+
+* Instructions for building sidecars
+* Distribution mechanism for 3rd party sidecars
+
+## Proposal
+
+We will introduce the following changes
+* enable the sidecar feature gate
+* setting to allow cluster administrators to whitelist sidecars for usage within the cluster
+
+### User Stories
+
+#### User wants to tweak kubevirt generated domain definition
+
+Users attempting to passthrough GPUs/vGPUs with more than 24G of memory to VM's booted in EFI mode, run into BAR resizing issues. 
+The default mmio aperture on in the firmware is not large enough to map the GPU/vGPU memory.
+This results in the OS being unable to leverage the hardware. This can be addressed by passing additional firmware arguments to tweak the mmio aperture size in the guest.
+
+Kubevirt does not directly expose this setting, and the only possible solution is to leverage a sidecar to patch the VM domain definition to add additional arguments.
+
+### User Experience In Detail
+Users will be expected to edit the VM definition as a yaml and apply the sidecar specfic annotations to their VM.
+
+For example the following annotations are needed to inject a sidecar and any additional arguments that the sidecar looks up on boot
+```
+    hooks.kubevirt.io/hookSidecars: '[{"args": ["--version", "v1alpha2"], "image":
+      "registry:5000/kubevirt/example-hook-sidecar:devel"}]'
+    smbios.vm.kubevirt.io/baseBoardManufacturer: Radical Edward
+```
+The VM will boot with the sidecar and apply the sidecar will apply the VM domain definition changes.
+
+### API changes
+
+#### Setting
+We will have a new setting `whitelisted-sidecars`
+
+```
+apiVersion: harvesterhci.io/v1beta1
```

Comment:
add an `eanbled` field? some users might not plan to control with while-list

Reply:
the plan is to use a `whitelist` to ensure users can control sidecars used cluster wide. With the sidecar the user will have access to core libvirt functionality not to mention devices attached to the launcher pod.

Reply:
the `whitelist` could be an option, to enable/disable whilelist control with a list;  if disabled, user can use any image

the consideration is:

some users tend to use terraform or others to operate the cluster, the explicit `whilelist` means addtional coding for them

Reply:
the idea does not conflict with the option to disable it, when users have the full control of cluster on hand  e.g. in an air-gapped env, the cluster users are from internal, the admin is free to disable this list and allow the trusted users to run the expected images.